.github/BUILD.md

```diff
@@ -13,81 +13,109 @@ If the answer is yes, please read on. Otherwise, make sure to head on over [to t
+### Familiarizing yourself with the code
+
+Umbraco is a .NET application using C#. The solution is broken down into multiple projects.  There are several class libraries. The `Umbraco.Web.UI` project is the main project that hosts the back office and login screen. This is the project you will want to run to see your changes.
+
+There are two web projects in the solution with client-side assets based on TypeScript, `Umbraco.Web.UI.Client` and `Umbraco.Web.UI.Login`.
+
+There are a few different ways to work locally when implementing features or fixing issues with the Umbraco CMS. Depending on whether you are working solely on the front-end, solely on the back-end, or somewhere in between, you may find different workflows work best for you.
+
+Here are some suggestions based on how we work on developing Umbraco at HQ.
 
-To build the front-end you'll need to open the command pallet (<kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>P</kbd>) and run `>Tasks: Run Task` followed by `Client Build`.
+### First checkout
 
-You can also run the tasks manually on the command line:
+When you first clone the source code, rebuild the whole solution via your IDE. You can then start the application via the IDE or the command line and should find everything across front and back-end is built and running.
 
 ```
-cd src\Umbraco.Web.UI.Client
-npm i
-npm run build:for:cms
+cd <solution root>\src\Umbraco.Web.UI
+dotnet run --no-build
 ```
 
-If you want to make changes to the UI, you can choose to run a front-end development server. To learn more please read the Umbraco.Web.UI.Client README.md [Run against a local Umbraco instance](../src/Umbraco.Web.UI.Client/.github/README.md#run-against-a-local-umbraco-instance) for more information.
+When the page loads in your web browser, you can follow the installer to set up a database for debugging. When complete, you will have an empty Umbraco installation to begin working with. You may also wish to install a [starter kit][https://marketplace.umbraco.com/category/themes-&-starter-kits] to ease your debugging.
 
-The login screen is a different frontend build, for that one you can run it as follows:
+### Back-end only changes
+
+If you are working on back-end only features, when switching branches or pulling down the latest from GitHub, you will find the front-end getting rebuilt periodically when you look to build the back-end changes. This can take a while and slow you down. So if for a period of time you don't care about changes in the front-end, you can disable this build step.
+
+Go to `Umbraco.Cms.StaticAssets.csproj` and comment out the following lines of MsBuild by adding a REM statement in front:
 
 ```
-cd src\Umbraco.Web.UI.Login
-npm i
-npm run dev
+REM npm ci --no-fund --no-audit --prefer-offline
+REM npm run build:for:cms
 ```
 
-If you just want to build the Login screen to `Umbraco.Web.UI` then instead of running `dev`, you can do: `npm run build`.
+Just be careful not to include this change in your PR.
```

Comment:
This is not going to run each time, right? Only when certain folders are missing. I have never commented these out and never had these commands run again.
Encouraging people to update `csproj` will just lead them to commit those changes 😉

Reply:
Well, in practice I have found it running when I "didn't want it", so I've disabled it myself.

It's a good thing to have for the first experience.  I.e. you just build in your IDE and everything will restore and build.  But I find it triggers if I've:

- Pulled down the latest from GitHub.  Could be there are some FE changes, but I don't really care about them if I've working on a BE only feature.
- Switching branches to review a PR with only BE changes.

And as it can take several minutes, and sometimes fails, this does get in the way of your flow.

Reply:
For sure, that's very unwanted. 

Might be a recent change then as I haven't experienced this. I see why it's there as a convenience but I don't like it 😅 
Anyway, fine to leave this `REM` hint in there then.

Reply:
Just because I was curious: indeed this came in in late November: https://github.com/umbraco/Umbraco-CMS/pull/17527/files#diff-096b40beec2cbe6b2055e325bb6ce85918006bc54fa003508311ffe53bfcfb6aR40-R52